internal/shortener/repositories/db_repository.go

```diff
@@ -126,3 +147,62 @@ func (repo *DatabaseRepository) GetByUserID(ctx context.Context, userID uuid.UUI
 
 	return shortURLs, nil
 }
+
+func (repo *DatabaseRepository) DeleteRecords(ctx context.Context, userID uuid.UUID, ids []string) error {
+	itemToDelete := &entities.ItemToDelete{
+		UserID:   userID,
+		ItemsIDs: ids,
+	}
+	go func() { repo.ToDelete <- itemToDelete }()
+	return nil
+}
+
+func (repo *DatabaseRepository) AccumulateRecordsToDelete() {
+	ticker := time.NewTicker(time.Millisecond * 500)
+	defer ticker.Stop()
+
+	localStorage := make(map[uuid.UUID][]string)
+
+	go func() {
+		for range ticker.C {
+			lockURLToDeleteStorage.Lock()
+			for userID, ids := range localStorage {
+				err := repo.DeleteRecordsForUser(context.Background(), userID, ids)
+				if err != nil {
+					repo.ToDelete <- &entities.ItemToDelete{
+						UserID:   userID,
+						ItemsIDs: ids,
+					}
+				}
+				delete(localStorage, userID)
+			}
+			lockURLToDeleteStorage.Unlock()
+		}
+	}()
+
+	for item := range repo.ToDelete {
+		lockURLToDeleteStorage.Lock()
+		urlsIDs, exist := localStorage[item.UserID]
+		if exist {
+			localStorage[item.UserID] = append(urlsIDs, item.ItemsIDs...)
+		} else {
+			localStorage[item.UserID] = item.ItemsIDs
+		}
+		lockURLToDeleteStorage.Unlock()
+	}
+}
+
+func (repo *DatabaseRepository) DeleteRecordsForUser(ctx context.Context, userID uuid.UUID, ids []string) error {
+	query, args, _ := sqlx.In(
+		"UPDATE short_urls SET is_active=false WHERE user_id = ? AND id IN (?)",
```

Comment:
Насколько я проблематику проброса параметров в SQL, в IN надо указывать список идентификаторов, а для того, чтобы пробросить слайс, как один параметр, необходимо задействовать ANY.
```suggestion
		"UPDATE short_urls SET is_active=false WHERE user_id = ? AND id=ANY(?)",
```
На 100% не помню, надо проверять. Поэтому не блокер.

Reply:
у меня там sqlx.In , она ловко работает с IN. Саму строку запроса анализировал, там все как нужно.